tests/test-sources/plugins/lazyloading/lz-n.nix

```diff
@@ -0,0 +1,307 @@
+{
+  lazy-load-neovim-plugin-configured =
+    { config, lib, ... }:
+    {
+      plugins = {
+        lz-n = {
+          enable = true;
+        };
+
+        neotest = {
+          enable = true;
+          lazyLoad = {
+            enable = true;
+            settings = {
+              keys = [
+                {
+                  __unkeyed-1 = "<leader>nt";
+                  __unkeyed-3 = "<CMD>Neotest summary<CR>";
+                  desc = "Summary toggle";
+                }
+              ];
+            };
+          };
+        };
+      };
+
+      assertions = [
+        {
+          assertion = (builtins.length config.plugins.lz-n.plugins) == 1;
+          message = "`lz-n.plugins` should have contained a single plugin configuration, but contained ${builtins.toJSON config.plugins.lz-n.plugins}";
+        }
+        {
+          assertion =
+            let
+              plugins = config.plugins.lz-n.plugins or [ ];
+              plugin = if builtins.length plugins > 0 then builtins.head plugins else null;
+              keys = if plugin != null && builtins.isList plugin.keys then plugin.keys else [ ];
+            in
```

Comment:
Hm... I wonder if it's worth having a small util file that can be imported by the relevant tests? This seems to be getting fairly repetitive?

Not a big deal though, if you're happy with it.